azurerm/internal/services/datashare/tests/resource_arm_data_share_account_test.go

```diff
@@ -0,0 +1,211 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datashare/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMDataShareAccount_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_share_account", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataShareAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataShareAccount_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataShareAccountExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "identity.0.principal_id"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "identity.0.tenant_id"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDataShareAccount_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_share_account", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataShareAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataShareAccount_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataShareAccountExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMDataShareAccount_requiresImport),
+		},
+	})
+}
+
+func TestAccAzureRMDataShareAccount_complete(t *testing.T) {
```

Comment:
The complete test should just have the complete step. a separate update test should go from basic->complete->basic, or basic->complete->complete-updated->basic